subsys/event_manager/event_manager.c

```diff
@@ -116,6 +116,15 @@ void __weak event_manager_trace_event_submission(const struct event_header *eh,
 {
 }
 
+void * __weak event_manager_alloc(const struct event_header *eh)
```

Comment:
Do we use this functions anywhere in current implementation?
Consider making them wrappers around `k_malloc` and `k_free` (allowing user to overwrite them with a custom implementation). Then we may also consider aligning function arguments.

Reply:
No we do not use it. Those functions are a result of requests by reviewers from Zephyr

Reply:
But we need to use it. This is the whole point :)

These should be used instead of kmalloc and kfree and should have the same interface.

```
void * __weak event_manager_alloc(size_t size)
{
   return kmalloc(size);
}

void __weak event_manager_free(void *addr)
{
   k_free(addr);
}
```
Please check if the interfaces are right.